experiments/surface_fluxes_perfect_model/sf_model.jl

```diff
@@ -0,0 +1,142 @@
+    gryanik_noniterative = false
+    return (
+        uf_params = uf_params,
+        scheme = scheme,
+        z0_momentum = z0_momentum,
+        z0_thermal = z0_thermal,
+        maxiter = maxiter,
+        tol_neutral = tol_neutral,
+        gryanik_noniterative = gryanik_noniterative,
+    )
+end
+
+"""
+    assemble_surface_conditions(thermo_params, prof_int, prof_sfc, z0m, z0b)
+
+Prepare the surface state structs for the surface fluxes model.
+"""
+function assemble_surface_conditions(
+    thermo_params,
+    prof_int,
+    prof_sfc,
+    z0m,
+    z0b,
+)
+    ts_sfc =
+        TD.PhaseEquil_ρTq(thermo_params, prof_sfc.ρ, prof_sfc.T, prof_sfc.q)
+    ts_int =
+        TD.PhaseEquil_ρTq(thermo_params, prof_int.ρ, prof_int.T, prof_int.q)
+
+    state_in = SF.StateValues(prof_int.z, (prof_int.u, prof_int.v), ts_int)
+    state_sfc = SF.StateValues(prof_sfc.z, (prof_sfc.u, prof_sfc.v), ts_sfc)
+    return SF.ValuesOnly(state_in, state_sfc, z0m, z0b)
+end
+
+"""
+    obtain_ustar(FT, x_inputs, model_config; return_ustar = false)
+
+Obtain the frictional velocity, ustar, from the surface fluxes model.
+"""
+function obtain_ustar(FT, x_inputs, model_config; return_ustar = false)
+
+    # dict containing theta
+    toml_dict = CP.create_toml_dict(
+        FT;
+        override_file = CP.merge_toml_files(model_config["toml"]),
+    )
+
+    # extract model inputs, x
+    (; profiles_sfc, profiles_int) = x_inputs
+
+    # generate stationary model parameters
+    (;
+        scheme,
+        z0_momentum,
+        z0_thermal,
+        maxiter,
+        tol_neutral,
+        gryanik_noniterative,
+    ) = generate_G_preliminaries(FT)
+
+    sch = scheme
+    param_set = SFPP.SurfaceFluxesParameters(toml_dict, UF.BusingerParams)
+    ustar_array = Array{FT}(
+        undef,
+        length(profiles_int),
+        length(z0_momentum),
+        length(z0_thermal),
+    )
+    @inbounds for (ii, prof_int) in enumerate(profiles_int)
+        prof_sfc = profiles_sfc[ii]
+        @inbounds for (kk, z0m) in enumerate(z0_momentum)
+            @inbounds for (ll, z0b) in enumerate(z0_thermal)
+                sc_states = assemble_surface_conditions(
+                    param_set.thermo_params,
+                    prof_int,
+                    prof_sfc,
+                    z0m,
+                    z0b,
+                )
+                sc = SF.surface_conditions(
+                    param_set,
+                    sc_states,
+                    sch;
+                    maxiter,
+                    tol_neutral,
+                    soltype = RS.VerboseSolution(),
+                    noniterative_stable_sol = gryanik_noniterative,
+                )
+
+                ustar_array[ii, kk, ll] = sc.ustar
+            end
+        end
+    end
+
+    # save ustar_array to file
+    JLD2.save(
+        joinpath(model_config["output_dir"], "model_ustar_array.jld2"),
+        Dict("ustar" => ustar_array),
+    )
```

Comment:
This use `save_object` and `load_object` to save the array directly instead of nesting it within a dict. This isn't a required change, but it is slightly cleaner!
```suggestion
    JLD2.save_object(
        joinpath(model_config["output_dir"], "model_ustar_array.jld2"),
        ustar_array,
    )
```

Reply:
Good idea! I've changed this.